radfield.cc

```diff
@@ -1138,11 +1138,25 @@ void reduce_estimators()
   MPI_Allreduce(MPI_IN_PLACE, J.data(), nonempty_npts_model, MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
   MPI_Allreduce(MPI_IN_PLACE, nuJ.data(), nonempty_npts_model, MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
 
+  // if constexpr (DETAILED_BF_ESTIMATORS_ON) {
+  //   for (ptrdiff_t nonemptymgi = 0; nonemptymgi < nonempty_npts_model; nonemptymgi++) {
+  //     MPI_Allreduce(MPI_IN_PLACE, &bfrate_raw[nonemptymgi * globals::bfestimcount], globals::bfestimcount,
+  //     MPI_DOUBLE,
+  //                   MPI_SUM, MPI_COMM_WORLD);
+  //   }
+  // }
+
   if constexpr (DETAILED_BF_ESTIMATORS_ON) {
     for (ptrdiff_t nonemptymgi = 0; nonemptymgi < nonempty_npts_model; nonemptymgi++) {
       MPI_Allreduce(MPI_IN_PLACE, &bfrate_raw[nonemptymgi * globals::bfestimcount], globals::bfestimcount, MPI_DOUBLE,
-                    MPI_SUM, MPI_COMM_WORLD);
+                    MPI_SUM, globals::mpi_comm_node);
```

Comment:
The result will be identical, but if you're not going to use the reduction results across the whole node (you overwrite it a few lines down for all ranks except for rank_in_node == 0) then you can avoid communicating the unused results by replacing MPI_Allreduce with MPI_Reduce:

`
MPI_Reduce(my_rank == 0 ? MPI_IN_PLACE : &bfrate_raw[nonemptymgi * globals::bfestimcount], &bfrate_raw[nonemptymgi * globals::bfestimcount], globals::bfestimcount, MPI_DOUBLE, MPI_SUM, 0, globals::mpi_comm_node);
`

Reply:
I ran with this version of the MPI_Allreduce on a cluster and ran into an error. I am running it again with the original code. I will investigate this error once the nt_MPI_Bcast error has been solved.